src/Mosaic/Mosaic.tsx

```diff
@@ -6,74 +6,68 @@ import { MosaicProps } from './Mosaic.interface'
 import {
   GridContainer,
   MosaicItem,
-  Item,
   FavouriteLayout,
   IconContainer,
 } from './Mosaic.style'
 
-export const Mosaic: React.FunctionComponent<MosaicProps> = (props) => {
-  const structure = React.useMemo((): React.ReactNode[] => {
-    const createMosaic = (items: React.ReactNode[], columnNumber: number) => {
-      const mosaic = (
-        <MosaicItem
-          data-items-length={items.length}
-          data-spacing={props.spacing}
-        >
-          {items
-            .map((item, i) => (
-              <Item
-                key={`column-${columnNumber}-item-${i}`}
-                data-rounded={props.rounded}
-              >
-                {item}
-              </Item>
-            ))
-            .reverse()}
-        </MosaicItem>
-      )
+export const Mosaic: React.FunctionComponent<MosaicProps> = ({
+  children,
+  spacing,
+  rounded,
+  columns,
+}) => {
+  const createMosaic = (items: React.ReactNode[], columnNumber: number) => {
+    const mosaic = (
+      <MosaicItem
+        data-items-length={items.length}
+        data-spacing={spacing}
+        data-rounded={rounded}
+      >
+        {items.map((item) => item).reverse()}
+      </MosaicItem>
+    )
 
-      return columnNumber === 1 ? (
-        <FavouriteLayout>
-          {mosaic}
-          <IconContainer>
-            <Icon icon="star" />
-          </IconContainer>
-        </FavouriteLayout>
-      ) : (
-        mosaic
-      )
-    }
+    return columnNumber === 1 ? (
+      <FavouriteLayout>
+        {mosaic}
+        <IconContainer>
```

Comment:
I don't think that the star should be in `Mosaic`, I think it should be managed outside of the `Mosaic` Layout